### PR TITLE
feat: add nft page

### PR DESCRIPTION
**Pull request by James0126**
## Describe your changes
 add nft page
## Demo Link
[Link](https://initia-celatone-frontend-git-feature-initianft-alleslabs.vercel.app/stone-12-1/nft-collections)

### CODE REVIEW

File: src/lib/amplitude/types.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ export enum AmpEvent {
   // NAVIGATE
   TO_OVERVIEW = "To Overview",
   TO_BLOCKS = "To Blocks",
+  TO_COLLECTIONS = "To NFTs",
```
Comment (evilpeach) [inline src/lib/amplitude/types.ts:32]:
```ts
TO_NFT_COLLECTIONS = "To NFT Collections"
```

File: src/lib/amplitude/types.ts [hunk 2]
```diff
@@ -53,9 +54,11 @@ export enum AmpEvent {
   TO_ALL_PROJECTS = "To All Public Projects",
   TO_ACCOUNT_DETAIL = "To Account Detail",
   TO_CONTRACT_DETAIL = "To Contract Detail",
+  TO_COLLECTION_DETAIL = "To Collection Detail",
```
Comment (evilpeach) [inline src/lib/amplitude/types.ts:57]:
```ts
TO_NFT_COLLECTION_DETAIL = "To NFT Collection Detail"
```

File: src/pages/collections/[collectionAddress]/[tab].tsx
Comment (songwongtp) [inline src/pages/collections/[collectionAddress]/[tab].tsx:1]: put everything under `/collections` and `/nft` route to under `/nfts`

File: src/pages/[network]/collections/[collectionAddress]/[tab].tsx
Comment (songwongtp) [inline src/pages/[network]/collections/[collectionAddress]/[tab].tsx:1]: put everything under `/collections` and `/nft` route to under `/nfts`

File: src/lib/pages/collection-list/components/Collections.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  return (
+    <Box>
+      <TextInput
+        variant="fixed-floating"
+        value={searchKeyword}
+        setInputState={setSearchKeyword}
+        placeholder="Search by collection name"
+        size="md"
+        mb="32px"
+      />
```
Comment (jennieramida) [inline src/lib/pages/collection-list/components/Collections.tsx:51]: Please change to

```jsx
<InputWithIcon
placeholder="Search with Collection Name"
value={searchKeyword}
autoFocus
onChange={(e) => setSearchKeyword(e.target.value)}
size={{ base: "md", md: "lg" }}
/>
```

File: src/lib/components/MobileGuard.tsx [hunk 4]
```diff
@@ -19,6 +19,8 @@ export const MobileGuard = ({ children }: MobileGuardProps) => {
     pathName.includes(`/blocks`) ||
     pathName.includes(`/projects`) ||
     pathName.includes(`/code`) ||
+    pathName.includes(`/collections`) ||
+    pathName.includes(`/nft`) ||
```
Comment (evilpeach) [inline src/lib/components/MobileGuard.tsx:23]: Let's merge this into `/nfts` route.

File: src/lib/pages/account-details/components/nfts/NFTsByCollection.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+  return (
+    <Stack spacing="24px">
+      <TextInput
+        variant="fixed-floating"
+        value={searchKeyword}
+        setInputState={setSearchKeyword}
+        placeholder="Search by name"
+        size="md"
+      />
```
Comment (jennieramida) [inline src/lib/pages/account-details/components/nfts/NFTsByCollection.tsx:84]: Please change to

```jsx
<InputWithIcon
placeholder="Search with NFT Name"
value={searchKeyword}
autoFocus
onChange={(e) => setSearchKeyword(e.target.value)}
size={{ base: "md", md: "lg" }}
/>
```

File: next.config.js [hunk 6]
```diff
@@ -44,6 +44,8 @@ const nextConfig = {
       "proposal",
       "project",
       "contract-list",
+      "nft-list",
+      "nft",
```
Comment (evilpeach) [inline next.config.js:48]: Just `nft`

File: src/lib/pages/collection-details/components/activities/index.tsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+      <Text fontSize="18px" fontWeight={600} fontFamily="Pilat Wide">
+        Activities
+      </Text>
+      <TextInput
+        variant="fixed-floating"
+        value={searchKeyword}
+        setInputState={setSearchKeyword}
+        placeholder="Search by Token Id"
+        size="md"
+      />
```
Comment (jennieramida) [inline src/lib/pages/collection-details/components/activities/index.tsx:62]: Please change to

```jsx
<InputWithIcon
placeholder="Search by Token ID"
value={searchKeyword}
autoFocus
onChange={(e) => setSearchKeyword(e.target.value)}
size={{ base: "md", md: "lg" }}
/>
```

File: src/lib/pages/collection-details/components/supplies/index.tsx [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+  return (
+    <Box mt="32px" gap="40px">
+      <TextInput
+        variant="fixed-floating"
+        value={searchKeyword}
+        setInputState={setSearchKeyword}
+        placeholder="Search by name"
+        size="md"
+        mb="32px"
+      />
```
Comment (jennieramida) [inline src/lib/pages/collection-details/components/supplies/index.tsx:59]: Please change to

```jsx
<InputWithIcon
placeholder="Search with NFT Name"
value={searchKeyword}
autoFocus
onChange={(e) => setSearchKeyword(e.target.value)}
size={{ base: "md", md: "lg" }}
/>
```

and add mt={8} to element below instead

File: src/lib/services/collectionService.ts [hunk 9]
```diff
@@ -0,0 +1,334 @@
...
+  const { chainConfig } = useCelatoneApp();
+
+  const queryFn = async () => {
+    return axios
+      .post<CollectionResponse>(chainConfig.indexer, {
+        query: getCollectionsPagination,
+        variables: { offset, pageSize, search },
+      })
+      .then(({ data: res }) =>
+        res.data.collections.map((collection) =>
```
Comment (songwongtp) [inline src/lib/services/collectionService.ts:64]: zCollectionResponseItem.array().parse(res.data.collections)

File: src/lib/services/collectionService.ts [hunk 10]
```diff
@@ -0,0 +1,334 @@
...
+        query: getCollectionActivitiesPagination,
+        variables: {
+          collectionAddress,
+          pageSize,
+          offset,
+          search,
+        },
+      })
+      .then(({ data: res }) =>
+        res.data.collection_transactions.map((tx) =>
```
Comment (songwongtp) [inline src/lib/services/collectionService.ts:251]: zActivitiesResponseItem.array().parse(res.data.collection_transactions)

File: src/lib/services/collectionService.ts [hunk 11]
```diff
@@ -0,0 +1,334 @@
...
+      .post<CollectionMutateEventsResponse>(chainConfig.indexer, {
+        query: getCollectionMutateEventsPagination,
+        variables: {
+          collectionAddress,
+          pageSize,
+          offset,
+        },
+      })
+      .then(({ data: res }) =>
+        res.data.collection_mutation_events.map((event) =>
```
Comment (songwongtp) [inline src/lib/services/collectionService.ts:289]: same

File: src/lib/services/collectionService.ts [hunk 12]
```diff
@@ -0,0 +1,334 @@
...
+export const useCollectionListByAddress = (accountAddress: HexAddr) => {
+  const { chainConfig } = useCelatoneApp();
+  const queryFn = async () => {
+    return axios
+      .post<CollectionListByAddressResponse>(chainConfig.indexer, {
+        query: getCollectionListByAddress,
+        variables: { accountAddress },
+      })
+      .then(({ data: res }) =>
+        res.data.collections.map((collection) =>
```
Comment (songwongtp) [inline src/lib/services/collectionService.ts:318]: same

File: src/lib/services/collectionService.ts [hunk 13]
```diff
@@ -0,0 +1,334 @@
...
+      .post<CollectionResponse>(chainConfig.indexer, {
+        query: getCollectionsPagination,
+        variables: { offset, pageSize, search },
+      })
+      .then(({ data: res }) =>
+        res.data.collections.map((collection) =>
+          zCollectionResponseItem.parse(collection)
+        )
+      );
+  };
```
Comment (evilpeach) [inline src/lib/services/collectionService.ts:68]: Let's develop a query function within collection.ts. This way, we can avoid importing the zod schema and response type directly into this file."

File: src/lib/services/nftService.ts [hunk 14]
```diff
@@ -0,0 +1,354 @@
...
+          collectionAddress,
+          limit: pageSize,
+          offset,
+          search: `%${search ?? ""}%`,
+        },
+      })
+      .then(({ data: res }) =>
+        res.data.nfts.map((nft) => zNFTTokenResponse.parse(nft))
+      );
+  };
```
Comment (evilpeach) [inline src/lib/services/nftService.ts:61]: Apply the same approach to this file as well.

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by songwongtp — CHANGES_REQUESTED** [review]

File: src/lib/components/MobileGuard.tsx (hunk 4 above)
Comment (evilpeach) [reply src/lib/components/MobileGuard.tsx:23]: As @songwongtp commented below.

**Comment by evilpeach** [comment]
Also, for consistency across the project, it's preferable to use `Nft` (capitalized) instead of `NFT` (uppercase). For instance, rename `NFTToken` to `NftToken`.

File: src/lib/pages/collection-list/components/Collections.tsx (hunk 3 above)
Comment (jennieramida) [reply src/lib/pages/collection-list/components/Collections.tsx:50]: please add margin top to collection list @James0126

Comment (James0126) [reply src/lib/pages/collection-list/components/Collections.tsx:50]: okay

File: src/lib/amplitude/types.ts [hunk 15]
```diff
@@ -67,6 +67,9 @@ export enum AmpEvent {
   TO_PUBLISH_MODULE = "To Publish Module",
   TO_DEPLOY_SCRIPT = "To Deploy Script",
   TO_MY_PUBLISHED_MODULES = "To My Published Modules",
+  TO_NFT_COLLECTIONS = "To NFT Collections",
+  TO_NFT_COLLECTION_DETAIL = "To NFT Collection Detail",
```
Comment (evilpeach) [inline src/lib/amplitude/types.ts:71]: Detail or Details?

Comment (James0126) [reply src/lib/amplitude/types.ts:71]: If you mean whether "Detail" or "Details" is correct, then "Details" seems more accurate.
i will change that.

Comment (James0126) [reply src/lib/amplitude/types.ts:71]: am I understanding this correctly? Most value use "Detail".

File: src/lib/pages/nft-details/components/attributes/AttributesModal.tsx [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+                borderRadius="8px"
+                p="8px 12px"
+                gap={1}
+                key={traitType}
+              >
+                <Text color="text.dark" fontWeight={700}>
+                  {traitType}
+                </Text>
+                <Text variant="body2" fontWeight={700}>
+                  {value}
```
Comment (evilpeach) [inline src/lib/pages/nft-details/components/attributes/AttributesModal.tsx:82]: Should we add default value for this?

**Review by evilpeach — APPROVED** [review]

**Review by songwongtp — APPROVED** [review]

**Pull request merged**